pkg/helpers/helpers.go

```diff
@@ -717,6 +719,20 @@ func SetRelatedResourcesStatuses(
 	}
 }
 
+func RemoveRelatedResourcesStatuses(
```

Comment:
this needs tests. And are you sure we need this? My understanding is everytime we will update the whole relatedResource list.

Reply:
Hmm, I think the existing func `SetRelatedResourcesStatuses` and this new func `RemoveRelatedResourcesStatuses` are to add/remove obj to/from the relatedResource when applying/deleting resource. They are running before `UpdateClusterManagerRelatedResourcesFn` (which is to update the whole list you mentioned here). 

Will add more testing code.

Reply:
Add some UT and add relatedResources check in integration testing.